src/Action.cpp

```diff
@@ -326,15 +324,11 @@ namespace gameController{
             }
         }
 
-        if(env->snitch->exists) {
-            if (this->env->snitch->position == this->target) {
-                if (INSTANCE_OF(this->actor, gameModel::Seeker)) {
-                    if (actionTriggered(env->config.gameDynamicsProbs.catchSnitch)) {
-                        actions.push_back(ActionResult::SnitchCatch);
-                        env->getTeam(actor)->score += SNITCH_POINTS;
-                    }
-                }
-            }
+        if(env->snitch->exists && this->env->snitch->position == this->target && INSTANCE_OF(this->actor, gameModel::Seeker)
```

Comment:
Vielleicht mit ein paar kleinen Hilfsvariableb
```suggestion
        const bool positionIsTarget = this->env->snitch->position == this->target;
        const bool isSeeker =  INSTANCE_OF(this->actor, gameModel::Seeker);
        const bool actionTriggered = actionTriggered(env->config.gameDynamicsProbs.catchSnitch);
        if(env->snitch->exists && positionIsTarget  && isSeeker && actionTriggered) {
```